### PR TITLE
[PropertyAccess] speed up accessing object properties

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28926, #29405
| License       | MIT
| Doc PR        | 

I propose to improve the performance of the `ObjectNormalizer` by not adding a new interface to the PropertyAccess component, but by adding some shortcut for cases where we know that we do not need to perform all checks. The added benefit is that this will not only speed up the `ObjectNormalizer` class, but will be available for every consumer of the `PropertyAccessor` without having to adapt to a new API.

TODO:

- [ ] confirm that these changes indeed introduce the same benefit as #29405 doing an actual benchmark

### CODE REVIEW

**Comment by fbourigault** [comment]
As this provide almost identical performance boost (~ 30% perf improvement). I would prefer merging this and do further profiling to improve serializer performance.

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php [hunk 1]
```diff
@@ -82,11 +82,16 @@ public function __construct(bool $magicCall = false, bool $throwExceptionOnInval
      */
     public function getValue($objectOrArray, $propertyPath)
     {
-        $propertyPath = $this->getPropertyPath($propertyPath);
-
         $zval = [
             self::VALUE => $objectOrArray,
         ];
+
+        if (\is_object($objectOrArray) && false === strpos((string) $propertyPath, '.') && false === strpos((string) $propertyPath, '[')) {
```
Comment (ostrolucky) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php]:
```suggestion
        if (\is_object($objectOrArray) && false === strpbrk((string) $propertyPath, '.[')) {
```

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php [hunk 2]
```diff
@@ -97,6 +102,22 @@ public function getValue($objectOrArray, $propertyPath)
      */
     public function setValue(&$objectOrArray, $propertyPath, $value)
     {
+        if (\is_object($objectOrArray) && false === strpos((string) $propertyPath, '.') && false === strpos((string) $propertyPath, '[')) {
```
Comment (ostrolucky) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php]:
```suggestion
        if (\is_object($objectOrArray) && false === strpbrk((string) $propertyPath, '.[')) {
```

**Comment by fbourigault** [comment]
Here is a profile of this PR (with @ostrolucky suggestions ; single `strpbrk` is slightly faster than two `strpos`) vs master : https://blackfire.io/profiles/compare/487eef43-6de6-4e82-954d-a08dc7c90e65/graph

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: Updated, though I wonder if using `strcspn()` and comparing that with `strlen()` would be better in that it doesn't required to copy a string.

**Review by fbourigault — APPROVED** [review]

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php (hunk 1 above)
Comment (fbourigault) [reply src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: According to https://blackfire.io/profiles/compare/f5b0113a-fcae-41cc-9b83-3bdd48c8bcf5/graph using `strpbrk ` is faster than `strcspn` + `strlen`.

**Review by dunglas — APPROVED** [review]

**Review by vudaltsov — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**